src/main/scala/chisel3/util/SRAM.scala

```diff
@@ -530,3 +618,40 @@ object SRAM {
     s"$rdPorts$wrPorts$rwPorts"
   }
 }
+
+/** Contains fields that map to a FIRRTL memory read port.
+  *
+  * @param readPort used to parameterize this class
+  */
+private[chisel3] final class FirrtlMemoryReader[T <: Data](readPort: MemoryReadPort[T]) extends Bundle {
```

Comment:
@jackkoenig should these bundles live here discreetly? The file is getting kind of long, so I was thinking about separating these out into a different file.

Reply:
I think they're fine here, but leave a comment about why they're private and mention that the write and readwrite bundles are missing `mask` (and explain why).